File: server/ManufacturersToYou/Domain/Manufacturer.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Manufacturer
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string City { get; set; }
        public string Description { get; set; }
        public string ImageUrl { get; set; }
        public virtual Category Category { get; set; }
        public virtual ICollection<Product> Products { get; set; }
    }
}

File: client/web-client-app/src/app/models/manufacturer.ts
export interface IManufacturersEnvelope {
    manufacturers: IManufacturer[],
    manufacturersCount: number
}

export interface IManufacturer {
    id: string,
    name: string,
    city: string,
    category: string,
    imageUrl: string,
    productsCount: number,
    products: IProduct[]
}

export interface IProduct {
    id: string,
    name: string,
    description: string,
    category: string,
    price: number,
    oldPrice: number,
    imageUrl: string,
    // city: string,
    photos: IPhoto[],
    features: IFeature[],
    comments: IComment[]
}

export interface IPhoto {
    id: string,
    url: string,
    isMain: boolean
}

export interface IFeature {
    key: string,
    value: string
}

export interface IComment {
    id: string;
    createdAt: Date;
    body: string;
    username: string;
    displayName: string;
    image: string;
}  